pkg/image/utilities/utilities.go

```diff
@@ -99,32 +100,43 @@ func GatherTargetImagesAndSources(systemChartsPath, chartsPath string, imagesFro
 		return ImageTargetsAndSources{}, fmt.Errorf("%s: %w", "could not write rancher-rke-k8s-versions.txt file", err)
 	}
 
-	externalImages := make(map[string][]string)
-	k3sUpgradeImages, err := ext.GetExternalImages(rancherVersion, data.K3S, ext.K3S, &semver.Version{
+	k8sVersion1_21_0 := &semver.Version{
 		Major: 1,
 		Minor: 21,
 		Patch: 0,
-	})
+	}
+
+	externalLinuxImages := make(map[string][]string)
+	externalWindowsImages := make(map[string][]string)
+
+	k3sUpgradeImages, err := ext.GetExternalImages(rancherVersion, data.K3S, ext.K3S, k8sVersion1_21_0, img.Linux /* unused for k3s */)
 	if err != nil {
 		return ImageTargetsAndSources{}, fmt.Errorf("%s: %w", "could not get external images for K3s", err)
 	}
 	if k3sUpgradeImages != nil {
-		externalImages["k3sUpgrade"] = k3sUpgradeImages
+		externalLinuxImages["k3sUpgrade"] = k3sUpgradeImages
 	}
 
 	// RKE2 Provisioning will only be supported on Kubernetes v1.21+. In addition, only RKE2
-	// releases corresponding to Kubernetes v1.21+ include the "rke2-images-all" file that we need.
-	rke2AllImages, err := ext.GetExternalImages(rancherVersion, data.RKE2, ext.RKE2, &semver.Version{
-		Major: 1,
-		Minor: 21,
-		Patch: 0,
-	})
+	// releases corresponding to Kubernetes v1.21+ include the "rke2-images-all.linux-amd64.txt" file that we need.
+	rke2LinuxImages, err := ext.GetExternalImages(rancherVersion, data.RKE2, ext.RKE2, k8sVersion1_21_0, img.Linux)
+	if err != nil {
+		return ImageTargetsAndSources{}, fmt.Errorf("%s: %w", "could not get external images for RKE2", err)
+
+	}
+	if rke2LinuxImages != nil {
+		externalLinuxImages["rke2LinuxAllImages"] = rke2LinuxImages
```

Comment:
Can you confirm that this hasn't broken the `image-sources.txt` files?

Reply:
```diff
@@ -13,39 +13,39 @@ rancher/fleet:v0.5.0 fleet:101.0.0+up0.5.0
 rancher/gitjob:v0.1.32 fleet:101.0.0+up0.5.0
 rancher/gke-operator:v1.1.5 rancher-gke-operator:101.0.0+up1.1.5
 rancher/grafana-grafana:7.1.5 rancher-monitoring:0.1.5
-rancher/hardened-calico:v3.22.2-build20220509 rke2All
-rancher/hardened-calico:v3.24.1-build20221011 rke2All
-rancher/hardened-cluster-autoscaler:v1.8.5-build20211119 rke2All
-rancher/hardened-cluster-autoscaler:v1.8.5-build20221011 rke2All
-rancher/hardened-cni-plugins:v1.0.1-build20220223 rke2All
-rancher/hardened-cni-plugins:v1.0.1-build20221011 rke2All
-rancher/hardened-coredns:v1.9.3-build20220613 rke2All
-rancher/hardened-coredns:v1.9.3-build20221011 rke2All
-rancher/hardened-dns-node-cache:1.21.2-build20211119 rke2All
-rancher/hardened-dns-node-cache:1.21.2-build20221011 rke2All
-rancher/hardened-etcd:v3.5.4-k3s1-build20220504 rke2All
-rancher/hardened-etcd:v3.5.4-k3s1-build20221011 rke2All
-rancher/hardened-flannel:v0.17.0-build20220317 rke2All
-rancher/hardened-flannel:v0.19.1-build20221011 rke2All
-rancher/hardened-ib-sriov-cni:v1.0.2-build20220419 rke2All
-rancher/hardened-k8s-metrics-server:v0.5.0-build20211119 rke2All
-rancher/hardened-k8s-metrics-server:v0.6.1-build20221011 rke2All
-rancher/hardened-kubernetes:v1.23.10-rke2r1-build20220817 rke2All
-rancher/hardened-kubernetes:v1.23.13-rke2r1-build20221013 rke2All
-rancher/hardened-kubernetes:v1.24.4-rke2r1-build20220817 rke2All
-rancher/hardened-kubernetes:v1.24.7-rke2r1-build20221013 rke2All
-rancher/hardened-multus-cni:v3.8-build20211104 rke2All
-rancher/hardened-multus-cni:v3.8-build20221011 rke2All
+rancher/hardened-calico:v3.22.2-build20220509 rke2LinuxAllImages
+rancher/hardened-calico:v3.24.1-build20221011 rke2LinuxAllImages
+rancher/hardened-cluster-autoscaler:v1.8.5-build20211119 rke2LinuxAllImages
+rancher/hardened-cluster-autoscaler:v1.8.5-build20221011 rke2LinuxAllImages
+rancher/hardened-cni-plugins:v1.0.1-build20220223 rke2LinuxAllImages
+rancher/hardened-cni-plugins:v1.0.1-build20221011 rke2LinuxAllImages
+rancher/hardened-coredns:v1.9.3-build20220613 rke2LinuxAllImages
+rancher/hardened-coredns:v1.9.3-build20221011 rke2LinuxAllImages
+rancher/hardened-dns-node-cache:1.21.2-build20211119 rke2LinuxAllImages
+rancher/hardened-dns-node-cache:1.21.2-build20221011 rke2LinuxAllImages
+rancher/hardened-etcd:v3.5.4-k3s1-build20220504 rke2LinuxAllImages
+rancher/hardened-etcd:v3.5.4-k3s1-build20221011 rke2LinuxAllImages
+rancher/hardened-flannel:v0.17.0-build20220317 rke2LinuxAllImages
+rancher/hardened-flannel:v0.19.1-build20221011 rke2LinuxAllImages
+rancher/hardened-ib-sriov-cni:v1.0.2-build20220419 rke2LinuxAllImages
+rancher/hardened-k8s-metrics-server:v0.5.0-build20211119 rke2LinuxAllImages
+rancher/hardened-k8s-metrics-server:v0.6.1-build20221011 rke2LinuxAllImages
+rancher/hardened-kubernetes:v1.23.10-rke2r1-build20220817 rke2LinuxAllImages
+rancher/hardened-kubernetes:v1.23.13-rke2r1-build20221013 rke2LinuxAllImages
+rancher/hardened-kubernetes:v1.24.4-rke2r1-build20220817 rke2LinuxAllImages
+rancher/hardened-kubernetes:v1.24.7-rke2r1-build20221013 rke2LinuxAllImages
+rancher/hardened-multus-cni:v3.8-build20211104 rke2LinuxAllImages
+rancher/hardened-multus-cni:v3.8-build20221011 rke2LinuxAllImages
```

It would probably affect people relying on the source.

Reply:
Is it possible to just reuse the `rke2All` key in both maps? instead of introducing os specific keys?

Reply:
Since we are dropping Windows support for now, I'm restoring this key.